compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionPreprocessing.cpp

```diff
@@ -131,14 +141,63 @@ struct FoldSuccessiveTensorInsertSliceOps
   }
 };
 
+struct GatherFusionPattern : public OpRewritePattern<linalg::YieldOp> {
```

Comment:
Can we make the pattern more robust? E.g., we can look at `linalg::GenericOp`, instead of `linalg.yeild`. In this context, you can get all the `tensor::ExtractOp` ops and replace them with what you have here. For the producer generic op, it is fusable only if

(1) It is a elementwise op , i.e., `linalg::isElementwise(producerOp) == true`
(2) And there are no tensor.extract op in the producer generic op.
(3) Perhaps we need to check that all the indexing maps are identity.

Reply:
Yes, added elementwise check. For (2), `isElementwise` has a check for extract in the call to `hasOnlyScalarElementwiseOp`. For (3), I think elementwise already checks to make sure it is permutation.